engine/table/src/main/java/io/deephaven/engine/table/impl/BlinkTableTools.java

```diff
@@ -52,122 +62,37 @@ public static Table blinkToAppendOnly(final Table blinkTable) {
      * @return An append-only in-memory table representing all data encountered in the blink table across all cycles
      *         till maximum row count
      */
-    public static Table blinkToAppendOnly(final Table blinkTable, long sizeLimit) {
+    public static Table blinkToAppendOnly(@NotNull final Table blinkTable, long sizeLimit) {
+        return blinkToAppendOnly(blinkTable, sizeLimit, null);
+    }
+
+    /**
+     * Convert a Blink Table to an in-memory append only table with a limit on maximum size. Any updates beyond that
+     * limit won't be appended to the table.
+     *
+     * @param blinkTable The input blink table
+     * @param sizeLimit The maximum number of rows in the append-only table
+     * @param memoKey A memo key to use to enable memoization (or {@code null} to disable memoization)
+     * @return An append-only in-memory table representing all data encountered in the blink table across all cycles
+     *         till maximum row count
+     */
+    public static Table blinkToAppendOnly(
+            final Table blinkTable,
+            final long sizeLimit,
+            @Nullable final Object memoKey) {
         if (sizeLimit < 0) {
             throw new IllegalArgumentException("Size limit cannot be negative, limit=" + sizeLimit);
         }
         final UpdateGraph updateGraph = blinkTable.getUpdateGraph();
         try (final SafeCloseable ignored = ExecutionContext.getContext().withUpdateGraph(updateGraph).open()) {
-            return internalBlinkToAppendOnly(blinkTable, sizeLimit);
-        }
-    }
+            final QueryTable coalesced = (QueryTable) blinkTable.coalesce();
 
-    private static Table internalBlinkToAppendOnly(final Table blinkTable, long sizeLimit) {
-        return QueryPerformanceRecorder.withNugget("blinkToAppendOnly", () -> {
-            if (!isBlink(blinkTable)) {
+            if (!isBlink(coalesced)) {
```

Comment:
I see the order has changed here - coalescing _first_ then checking blink. Does it make more sense to keep the check first?

More generally, are we really using coalesce() b/c we need to make sure manage livenessscope stack stuff? If so, are we maybe abusing coalesce in this context b/c we already know a blinktable is coalesced? (And maybe, we should have a more appropriate method on Table for this situation?)

Reply:
Good catch. That is my mistake. I needed the QueryTable to invoke `getResult` and I got the order wrong.